### PR TITLE
Send overall (rather than per-stage) completion ratio to Studio

### PR DESCRIPTION
**Pull request by thomshutt**

### CODE REVIEW

File: clients/callback_client.go [hunk 1]
```diff
@@ -86,21 +90,54 @@ func (c CallbackClient) SendTranscodeStatusError(callbackURL, errorMsg string) e
...
+	if currentStageCompletionRatio < 0 {
+		currentStageCompletionRatio = 0
+	}
+	if currentStageCompletionRatio > 1 {
+		currentStageCompletionRatio = 1
+	}
+
+	// Define the "base" numbers - e.g the overall ratio we start each stage at
+	var TranscodeStatusPreparingBase float64 = 0.0
+	var TranscodeStatusPreparingTranscodingBase float64 = 0.4
```
Comment (victorges) [inline clients/callback_client.go:107]: Should it be only "Transcoding"?
```suggestion
	var TranscodeStatusTranscodingBase float64 = 0.4
```

File: clients/callback_client.go [hunk 2]
```diff
@@ -86,21 +90,54 @@ func (c CallbackClient) SendTranscodeStatusError(callbackURL, errorMsg string) e
...
+	var TranscodeStatusCompletedBase float64 = 1
+
+	switch status {
+	case TranscodeStatusPreparing:
+		return TranscodeStatusPreparingBase + (currentStageCompletionRatio * (TranscodeStatusPreparingTranscodingBase - TranscodeStatusPreparingBase))
+	case TranscodeStatusPreparingCompleted:
+		return TranscodeStatusPreparingTranscodingBase
+	case TranscodeStatusTranscoding:
+		return TranscodeStatusPreparingTranscodingBase + (currentStageCompletionRatio * (TranscodeStatusCompletedBase - TranscodeStatusPreparingTranscodingBase))
```
Comment (victorges) [inline clients/callback_client.go:116]: Another way of implementing this, to have the more complex math only in 1 place, would be to have 2 vars "start" and "end" and force the `currentStageCompletionRatio` to be `1` on the "Completed" steps (but use their same start/end as the non-completed). Sth like
```go
	switch status {
	case TranscodeStatusPreparing, TranscodeStatusPreparingCompleted:
		start, end = 0, 0.4
	case TranscodeStatusTranscoding, TranscodeStatusTranscodingCompleted:
		start, end = 0.4, 1
        }
        if (status == TranscodeStatusPreparingCompleted || status == TranscodeStatusTranscodingCompleted) {
                currentStageCompletionRatio = 1
        }
```


Then there could be a single progress-scaling function like this: https://github.com/livepeer/task-runner/blob/4d1ae1768c01e979969bf7f00b81267db05e299c/task/progress.go#L66

Certainly not a required change, just tripping on the impl. Could make sense when we start adding more steps here. WDYT?

**Review by victorges — APPROVED** [review]
Cool!

File: clients/callback_client.go (hunk 1 above)
Comment (thomshutt) [reply clients/callback_client.go:107]: The "Base" was trying to signify that this is the lower limit / floor / whatever, open to suggestions for better names though. Maybe "Floor"?

Comment (thomshutt) [reply clients/callback_client.go:107]: Just realised what you were actually pointing out 👓

File: clients/callback_client.go (hunk 2 above)
Comment (thomshutt) [reply clients/callback_client.go]: I like that a lot!

File: clients/callback_client.go [hunk 3]
```diff
@@ -86,21 +90,54 @@ func (c CallbackClient) SendTranscodeStatusError(callbackURL, errorMsg string) e
...
+	if status == TranscodeStatusPreparingCompleted || status == TranscodeStatusCompleted {
+		currentStageCompletionRatio = 1
+	}
+
+	switch status {
+	case TranscodeStatusPreparing, TranscodeStatusPreparingCompleted:
+		return scaleProgress(currentStageCompletionRatio, 0, 0.4)
+	case TranscodeStatusTranscoding, TranscodeStatusCompleted:
+		return scaleProgress(currentStageCompletionRatio, 0.4, 1)
+	}
```
Comment (victorges) [inline clients/callback_client.go:115]: Nice!

**Pull request merged**